python/whylogs/api/whylabs/session_manager.py

```diff
@@ -0,0 +1,113 @@
+import os
+import uuid
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Optional, Union
+
+import keyring
+from whylabs_client.api.sessions_api import (
+    CreateSessionRequest,
+    CreateSessionResponse,
+    SessionsApi,
+)
+from whylabs_client.api_client import ApiClient, Configuration
+
+from .auth_file import get_auth_file_path, get_configparser_object
+from .variables import Variables
+
+_auth_path = get_auth_file_path(auth_path=Path(f"{Path.home()}/.whylabs/auth.ini"))
+DEFAULT_WHYLABS_HOST = "https://songbird.development.whylabsdev.com"
+
+
+class GuestSession:
+    def __init__(self, session_id: str):
+        self.session_id = session_id
+        Variables.set_variable_to_config_file(key="session_id", value=self.session_id, auth_path=_auth_path)
+
+
+@dataclass
+class UserSession:
+    org_id: str
+    dataset_id: str
+    api_key: str
+
+    def __post_init__(self):
+        Variables.set_variable_to_config_file(key="org_id", value=self.org_id, auth_path=_auth_path)
+        Variables.set_variable_to_config_file(key="dataset_id", value=self.dataset_id, auth_path=_auth_path)
+        keyring.set_password("whylabs", "api_key", self.api_key)
```

Comment:
Does this end up prompting them for a password when we call `get`? And is this the right spot to put some of the `if_notebook` checks? I assume we'll have to do that for some of the UX for entering settings using `input()`.